catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    /**
+     * Return the directory being observed.
+     *
+     * @return the directory being observed
+     */
+    public String getDirectory() {
+        return rootEntry.getLocation();
+    }
+
+    /**
+     * Add a file system listener.
+     *
+     * @param listener The file system listener
+     */
+    void addListener(final EntryAlterationListener listener) {
+        if (listener != null) {
+            listeners.add(listener);
+        }
+    }
+
+    /**
+     * Remove a file system listener.
+     *
+     * @param listener The file system listener
+     */
+    void removeListener(final EntryAlterationListener listener) {
+        if (listener != null) {
+            while (listeners.remove(listener)) {
+            }
+        }
+    }
+
+    /**
+     * Returns the set of registered file system listeners.
+     *
+     * @return The file system listeners
+     */
+    public Iterable<EntryAlterationListener> getListeners() {
+        return listeners;
+    }
+
+    /**
+     * Initialize the observer. Note this means no events will be fired for
+     * files that already existed before monitoring started.
+     *
+     * @param sardine
+     */
+    public void initialize(Sardine sardine) {
+        this.sardine = sardine;
+        try {
+            DavResource davResource = sardine.list(rootEntry.getLocation())
+                    .get(0);
+            rootEntry.refresh(davResource);
+            final DavEntry[] children = doListFiles(rootEntry.getLocation(), rootEntry);
+            rootEntry.setChildren(children);
+        } catch (IOException e) {
+            // probably means root location was inaccessible
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Final processing.
+     *
+     * @throws Exception if an error occurs
+     */
+    public void destroy() throws Exception {
+    }
+
+    /**
+     * Check whether the file and its chlidren have been created, modified or deleted.
+     */
+    void checkAndNotify(Sardine sardine) {
+        this.sardine = sardine;
+        if (rootEntry.remoteExists(sardine)) {
+            checkAndNotify(rootEntry, rootEntry.getChildren(), listFiles(rootEntry.getLocation()));
+        } else if (rootEntry.isExists()) {
+            // would remove all metacards if connectivity is lost, but tough to distinguish
+            // from the hierarchy not existing remotely
+            checkAndNotify(rootEntry, rootEntry.getChildren(), EMPTY_RESOURCES);
+        } else {
+            // Didn't exist and still doesn't
+        }
+    }
+
+    /**
+     * Compare two file lists for files which have been created, modified or deleted.
+     *
+     * @param parent   The parent entry
+     * @param previous The original list of files
+     * @param incoming The current list of files
+     */
+    private void checkAndNotify(final DavEntry parent, final DavEntry[] previous,
+            final DavResource[] incoming) {
+        int c = 0;
```

Comment:
Personally I like the convention of single character names for indexes that are incremented in a loop.  Given the number of times that index is referenced, I don't think renaming it would make the code more readable.